picard/formats/id3.py

```diff
@@ -684,6 +727,31 @@ def format_specific_metadata(self, metadata, tag, settings=None):
 
         return values
 
+    def _parse_sylt_text(self, text):
+        lrc_lyrics = []
+        previous_line_ended = True
+        for lyrics, milliseconds in text:
+            minutes = milliseconds // (60 * 1000)
+            seconds = (milliseconds % (60 * 1000)) // 1000
+            hundredths = (milliseconds % 1000) // 10
+            if previous_line_ended:
+                lrc_lyrics.append(f"[{minutes:02d}:{seconds:02d}.{hundredths:02d}]{lyrics}")
```

Comment:
Thank you for the thorough explanation!

I tried estimating the line-timestamp like you suggested by looking ahead. But I'm not sure how the estimation should be done. As of now I just take the time in the middle of two SYLT frames, but that does not make much sense, since all line-timestamps within a SYLT frame will show the same time.

Maybe it's best to evenly spread the difference between two SYLT frames among all line-timestamps. For example "<00:00.000>Ahhhhhhhhhhhhhhhhhhhh-\nhhh-\nhhh-\nhhhh<00:40.000>men" becomes:
```
[00:00.000]<00:00.000>Ahhhhhhhhhhhhhhhhhhhh-
[00:10.000]hhh-
[00:20.000]hhh-
[00:30.000]hhh-
[00:40.000]<00:40.000>men
```
Then there's also the issue of the last SYLT frame. Since there is no successive frame, it's hard to estimate the timestamps. As of now the code just forgoes the estimation for the last SYLT frame and copies the timestamp. As an example, "<00:10.000>Ahhhhhhhhhhhhhhhhhhhh-\nhhh-\nhhh-\nhhhh" becomes:
```
[00:10.000]<00:10.000>Ahhhhhhhhhhhhhhhhhhhh-
[00:10.000]hhh-
[00:10.000]hhh-
[00:10.000]hhh-
```
Hopefully I was clear enough!

Reply:
I think the first example is better than the second, but even better:
```
[00:00.000]<00:00.000>Ahhhhhhhhhhhhhhhhhhhh- (23 chars in \n)
[00:xx.000]hhh- (5 chars inc \n)
[00:xx.000]hhh- (5 chars inc \n)
[00:xx.000]hhh- (4 chars)
[00:40.000]<00:40.000>men

So there are 37 chars spread across 40secs, so you can estimate this as 40/37 s/char and so 23*40/37 = 25s for the first line, then 5*40/37 = 5s for each of the remaining lines resulting in:
[00:00.000]<00:00.000>Ahhhhhhhhhhhhhhhhhhhh-
[00:25.000]hhh-
[00:30.000]hhh-
[00:35.000]hhh-
[00:40.000]<00:40.000>men 
```

Reply:
Where the multi-line frame is the last frame and we have no end time on the lyrics, you can use the track time to estimate the length of the subframe. So in your example, if the track is 40s long you would get the same spread of times.

Reply:
Thank you for the idea, this makes much more sense now.